### PR TITLE
feat: add read-only team and usergroups commands

### LINKED ISSUE #139: Add team and usergroups commands for workspace and user-group management
### What

Two new command groups:

- `slackcli team info` — read the workspace's own metadata (name, domain, ID, verification) via `team.info`.
- `slackcli usergroups` — list, read, and manage Slack **User Groups** ("subteams"): `list`, `read`, `create`, `update`, `add`/`remove` members, `enable`/`disable`.

Every `usergroups` subcommand accepts a group by ID, `@handle`, or exact name, plus the standard `--workspace`, `--json`, and a `--team <workspace-id>` for enterprise-grid scoping.

### Why

`slackcli` can already read conversations, messages, search, and emoji, but there is no way to see or manage the workspace's **user groups** — the mentionable `@team` handles that map real org structure. This is a common automation need (onboarding/offboarding a person across their groups, auditing group membership, scripting group creation) and today requires the Slack web UI or a custom app. Both `team.info` and the `usergroups.*` methods work under `slackcli`'s existing dual-auth model, so no new auth surface is introduced.

### How (optional)

Follows the existing `docs/development/adding-a-command.md` pattern: types in `src/types`, API calls routed through `SlackClient.request()` (both auth types), logic in `src/lib/usergroups.ts`, formatters in `src/lib/formatter.ts`, thin command files, and unit tests.

Slack's `usergroups.users.update` replaces a group's **entire** member list, so `add`/`remove` are implemented as a safe read-modify-write (preserve concurrent members, no-op when unchanged, refuse to empty the last member). On an Enterprise Grid, writes require a `--team` workspace ID — verified live, without which Slack returns `target_team_must_be_specified_in_org_context`.

**Out of scope:** adding/removing users to the *workspace itself* (`admin.users.*`) — that needs an admin/enterprise token that browser session tokens can't provide.

I have a signed, tested branch ready (12 files, +1122, full lifecycle verified live against a throwaway group) and will open the PR once this is triaged with `ready-for-pr`.

### PR DESCRIPTION
**Pull request by tmack8001**
## Linked issue

Part of #139

## Summary

Adds the **read-only** surface of two new command groups, mirroring Slack's own API naming:

- `slackcli team info` — workspace (team) metadata (name, domain, ID, verification) via `team.info`.
- `slackcli usergroups list` — the workspace's User Groups ("subteams"), sorted, with handle / member count / enabled state (`--include-disabled`).
- `slackcli usergroups read <group>` — one group and its members, resolving member IDs to names. `<group>` is an id, `@handle`, or exact name.

Every command takes `--workspace`, `--json`, and `--team <workspace-id>` for Enterprise Grid scoping. All calls route through the read methods `team.info` / `usergroups.list` / `usergroups.users.list`, which work under both auth types (standard `xoxb`/`xoxp` and browser `xoxc`/`xoxd`), so no new auth surface is introduced.

**Read/write split (per the discussion on #139):** this is the first of two PRs — read-only here, and a follow-up PR for the write verbs (`create`/`update`/`add`/`remove`/`enable`/`disable`, with a `--yes` confirmation gate) once this merges. Keeping the read side separate makes it low-risk and independently reviewable.

One design note: `resolveUsergroup` honours a raw `S…` id directly rather than requiring a `usergroups.list` hit — on an Enterprise Grid a group scoped to a member workspace does not always appear in the org-context list, so an id-only lookup would otherwise fail.

**Tests:** `src/lib/usergroups.test.ts` — 11 cases covering `normalizeUsergroups` (field mapping + sort + empty input), `resolveUsergroup` (by id / @handle / exact name / no-match / raw-S-id-absent-from-list), `isUsergroupEnabled`, and `fetchUsergroupMembers` (id→name resolution preserving order, empty-group short-circuit). Full suite: 433 pass. Verified live against a real workspace: `team info` and `usergroups list`/`read` work, and `usergroups --help` shows only `list` and `read` (no write verbs).

## Checklist

- [x] The linked issue is open and carries the `ready-for-pr` label (constitution §1–2)
- [x] Change is focused on the linked issue — no unrelated edits
- [x] Tests added/updated, including edge cases (constitution §4)
- [x] `bun run type-check` and `bun test` pass locally
- [x] Pre-commit hooks are installed and passing — no `--no-verify`, no `SKIP=` (constitution §5)
- [x] All commits are signed (constitution §5 — unsigned commits make the PR unmergeable)
- [x] Documentation in `docs/` updated, added, or deleted as needed (constitution §7)
- [x] No tokens, credentials, or user data handled insecurely

### CODE REVIEW

File: src/commands/team.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+import type { SlackTeam } from '../types/index.ts';
+
+export function createTeamCommand(): Command {
+  const team = new Command('team')
+    .description('View the workspace (team) itself');
+
+  team
+    .command('info')
+    .description('Show workspace name, domain, and ID')
+    .option('--workspace <id|name>', 'Workspace to use')
```
Comment (vibexp-agent) [inline src/commands/team.ts:14]: The description says every command takes `--team <workspace-id>` for Enterprise Grid scoping, but `team info` does not declare it and `getTeamInfo()` sends no `team` param. `team.info` does accept one ("Team to get info about; if omitted, will return information about the current team"), so on a grid this always reports the token's own workspace.

File: src/lib/usergroups.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+import type { SlackUsergroup, UsergroupMember } from '../types/index.ts';
+
+type ProgressOptions = { onProgress?: (message: string) => void };
+type ScopeOptions = { teamId?: string };
+type LibOptions = ProgressOptions & ScopeOptions;
+
+// A user group is "enabled" when Slack has not soft-deleted it. Slack encodes
+// that as date_delete === 0 (a disabled group carries the unix time it was
+// disabled). Centralised so the formatter and commands agree on the rule.
+export function isUsergroupEnabled(group: Pick<SlackUsergroup, 'date_delete'>): boolean {
```
Comment (vibexp-agent) [inline src/lib/usergroups.ts:12]: This says it is centralised "so the formatter and commands agree on the rule", but `formatter.ts` defines its own `usergroupEnabled` inline and nothing outside `usergroups.test.ts` calls this one. Two copies of a rule that must agree, one of them dead. Pick one.

**Review by vibexp-agent — CHANGES_REQUESTED** [review]
Two inline notes: `team info` has no `--team` despite the description saying every command takes it, and `isUsergroupEnabled` is duplicated rather than shared. Separately, `Part of #139` does not populate `closingIssuesReferences`, so the linked-issue check will go red as written.

File: src/commands/team.ts (hunk 1 above)
Comment (tmack8001) [reply src/commands/team.ts:14]: Fixed in 59c108b. `team info` now declares `--team <workspace-id>` and `getTeamInfo({ team })` sends `team` to `team.info` when set, so on an Enterprise Grid you can scope to a specific member workspace instead of always getting the token's own. Documented the option in `docs/user-guide/team.md` too. Type-check + 433 tests green.

File: src/lib/usergroups.ts (hunk 2 above)
Comment (tmack8001) [reply src/lib/usergroups.ts:12]: Fixed in 59c108b. Removed the inline `usergroupEnabled` duplicate in `formatter.ts` and made both formatter call sites import and use this `isUsergroupEnabled` — so there is now one definition, and it is live (the formatter is a real caller, not just the test). Updated the comment here to match.

**Comment by tmack8001** [comment]
On the linked-issue note: the empty `closingIssuesReferences` is intentional, not an oversight.

Per your ruling on #139, this ships as two independent PRs against `main` — read-only first (this PR), then the write PR after this merges. To keep #139 open until *both* halves land, this PR deliberately uses `Part of #139` (references, does not close), and the follow-up **write** PR will carry `Closes #139`. If this PR closed #139 on merge, the issue would be gone while the write surface it also covers is still unshipped.

So the linked-issue check going red here is expected for the read half. If your workflow treats a red linked-issue check as a hard merge gate regardless, let me know and I can switch this to `Closes #139` and have the write PR re-reference it instead — that is your call on which PR should own the close.

**Comment by tmack8001** [comment]
Heads up for maintainer visibility — CI isn't actually running on this PR yet.

The `test` check shows `Expected — Waiting for status to be reported`, and there are **3 workflows awaiting approval** with "This workflow requires approval from a maintainer." This is GitHub's default gate for a pull request from a **first-time / outside contributor's fork** — the workflows sit in `action_required` and never start until a maintainer clicks **Approve and run** on the PR's Checks tab (or the "X workflows awaiting approval" banner).

Only `check-signatures` has run so far (it's not gated); everything else is blocked behind that approval, so there is currently no real test signal on the branch.

Could you **approve the workflow runs** when you get a chance? Once they run I'll address anything they surface. (If you'd rather relax this for repeat contributors, the repo setting is *Settings → Actions → General → Fork pull request workflows from outside collaborators* — but that's entirely your call; approving this PR's runs is all that's needed to unblock it.)

**Review by vibexp-agent — COMMENTED** [review]
Both are fixed. `--team` is now on all three subcommands and reaches `team.info` / `team_id` on both usergroups calls, and `isUsergroupEnabled` has one definition with the formatter as a live caller. 433 pass and type-check clean merged with main. On whether the read half or the write half should carry `Closes #139`: that is Shaharia's call, I have flagged it. Thanks for splitting this read-only.

**Comment by vibexp-agent** [comment]
Keeping `Part of #139` as you have it, and I have added `no-issue-needed` so the linked-issue check stops failing on a split that is deliberate. #139 stays open for the write half, which should carry `Closes #139`.

**Review by vibexp-agent — DISMISSED** [review]
Verified merged with main: type-check clean, 433 pass. Keeping the read/write split as you designed it. Merging.

**Review by vibexp-agent — APPROVED** [review]
Merged main in and resolved the additive conflicts with the emoji group from #149. All checks green. Merging.
